src/pyosrd/infra/helpers/station_builder.py

```diff
@@ -0,0 +1,153 @@
+
+    infra_builder.add_point_switch(
+        track_in.end(),
+        t1.begin(),
+        t2.begin(),
+        label=name+'.DVG',
+    )
+    infra_builder.add_point_switch(
+        track_out.begin(),
+        t1.end(),
+        t2.end(),
+        label=name+'.CVG',
+    )
+
+    # D0
+    track_in.add_detector(
+            label=name+'.D0',
+            position=track_in.length - 20,
+        )
+
+    # D1/S1
+    t1.add_detector(
+            label=name+'.D1',
+            position=400,
+        )
+    s = t1.add_signal(
+                    380,
+                    Direction.START_TO_STOP,
+                    is_route_delimiter=True,
+                    label=name+'.S1',
+                )
+    s.add_logical_signal("BAL", settings={"Nf": "true"})
+
+    # D2/S2
+    t1.add_detector(
+            label=name+'.D2',
+            position=800,
+        )
+    s = t1.add_signal(
+                    820,
+                    Direction.STOP_TO_START,
+                    is_route_delimiter=True,
+                    label=name+'.S2',
+                )
+    s.add_logical_signal("BAL", settings={"Nf": "true"})
+
+    # D3/S3
+    t2.add_detector(
+            label=name+'.D3',
+            position=400,
+        )
+    s = t2.add_signal(
+                    380,
+                    Direction.START_TO_STOP,
+                    is_route_delimiter=True,
+                    label=name+'.S3',
+                )
+    s.add_logical_signal("BAL", settings={"Nf": "true"})
+
+    # D4/S4
+    t2.add_detector(
+            label=name+'.D4',
+            position=800,
+        )
+    s = t2.add_signal(
+                    820,
+                    Direction.STOP_TO_START,
+                    is_route_delimiter=True,
+                    label=name+'.S4',
+                )
+    s.add_logical_signal("BAL", settings={"Nf": "true"})
+
+    # D5
+    track_out.add_detector(
+            label=name+'.D5',
+            position=20,
+        )
+
+    s1 = infra_builder.add_operational_point(label=name+'.s1')
+    s2 = infra_builder.add_operational_point(label=name+'.s2')
+    s1.add_part(t1, 500)
+    s2.add_part(t2, 500)
+
+    return track_out
+
+
+def build_N_dvg_station_cvg(
+    infra_builder: InfraBuilder,
+    track_in: TrackSection,
+    base_name: str,
+    N: int
+) -> TrackSection:
+    """Create a serie of N stations (see build_dvg_station_cvg for details).
+
+    All stations are created and connected one after the other. Their names
+    are based on the base_name parameter with a numbered suffix.
+
+    Args:
+        infra_builder (InfraBuilder): The InfraBuilder used to create
+        all parts of the infra
+        track_in (TrackSection): The track the first station will be
+        connected to
+        base_name (str): The name to be used for all stations
+        N (int): The number of station do be created
+
+    Returns:
+        TrackSection: Return the last track section of all the stations.
+    """
+    track = track_in
```

Comment:
I am pretty sure list comprehension cannot be used here because :
- We are not building any list
- We are using the output of every iteration for the next one

I could not find any example on various sources on how to use list comprehension for anything else than building a "simple" list.